File: examples/sync/bingx_public.py
import dcex


def main() -> None:
    client = dcex.bingx()

    ticker = client.get_ticker(product_symbol="BTC-USDT-SWAP")
    print(ticker)

    orderbook = client.get_spot_orderbook(product_symbol="BTC-USDT-SPOT", limit=5)
    print(orderbook)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/bingx_public.rs
use std::time::Duration;

use dcex::exchanges::bingx::BingxClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = BingxClient::public(Duration::from_secs(10))?;
    let response = client.get_orderbook("BTC-USDT-SWAP").limit(5).await?;
    println!("{}", response.data);
    Ok(())
}
